features/dryrun_mode.feature

```diff
@@ -0,0 +1,134 @@
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+    """
+    var cucumberSteps = function() {
+      this.When(/^this step passes$/, function(callback) { callback(); });
+    };
+    module.exports = cucumberSteps;
+    """
+    When I run cucumber.js with `-f progress features/a.feature --dry-run --strict`
+    Then it outputs this text:
+    """
+    -
+
+    1 scenario (1 skipped)
+    1 step (1 skipped)
+
+    """
+    And the exit status should be 0
+
+  Scenario: Undefined scenario with undefined step with --dry-run --strict
+    When I run cucumber.js with `-f progress features/a.feature --dry-run --strict`
+    Then it outputs this text:
+    """
+    U
+
+    1 scenario (1 undefined)
+    1 step (1 undefined)
+
+    You can implement step definitions for undefined steps with these snippets:
+
+    this.Given(/^this step passes$/, function (callback) {
+      // Write code here that turns the phrase above into concrete actions
+      callback.pending();
+    });
+    """
+    And the exit status should be 1
+
+  Scenario: Skipped scenario with undefined step with --dry-run
+    When I run cucumber.js with `-f progress features/a.feature --dry-run`
+    Then it outputs this text:
+    """
+    -
+
+    1 scenario (1 skipped)
+    1 step (1 skipped)
+
+    """
+    And the exit status should be 0
+
+  Scenario: Skipped Scenario with pending step with --dry-run
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+    """
+    var cucumberSteps = function() {
+      this.Given(/^this step passes$/, function(callback) { callback.pending(); });
+    };
+    module.exports = cucumberSteps;
+    """
+    When I run cucumber.js with `-f progress features/a.feature --dry-run`
+    Then it outputs this text:
+    """
+    -
+
+    1 scenario (1 skipped)
+    1 step (1 skipped)
+    """
+    And the exit status should be 0
+
+  Scenario: Skipped scenario with implemented step with -d
+    Given a file named "features/step_definitions/cucumber_steps.js" with:
+    """
+    var cucumberSteps = function() {
+      this.When(/^this step passes$/, function(callback) { callback(); });
+    };
+    module.exports = cucumberSteps;
+    """
+    When I run cucumber.js with `-f progress features/a.feature -d`
+    Then it outputs this text:
+    """
+    -
+
+    1 scenario (1 skipped)
+    1 step (1 skipped)
+
+    """
+    And the exit status should be 0
+
+  Scenario: Undefined scenario with undefined step with -d -S
+    When I run cucumber.js with `-f progress features/a.feature -d -S`
+    Then it outputs this text:
+    """
+    U
+
+    1 scenario (1 undefined)
+    1 step (1 undefined)
+
+    You can implement step definitions for undefined steps with these snippets:
+
+    this.Given(/^this step passes$/, function (callback) {
+      // Write code here that turns the phrase above into concrete actions
+      callback.pending();
+    });
+    """
+    And the exit status should be 1
```

Comment:
I don't think this test case is needed.
strict has its own test case that -S and --strict is synonymous.